packages/shared/src/regex.ts

```diff
@@ -2,6 +2,7 @@ export const emailRegEx = /^\S+@\S+\.\S+$/;
 export const phoneRegEx = /^\d+$/;
 export const usernameRegEx = /^[A-Z_a-z]\w*$/;
 export const passwordRegEx = /^.{6,}$/;
-export const redirectUriRegEx = /^https?:\/\//;
+export const redirectUriRegEx = /^https?:\/\/.*/;
+export const redirectNativeRegEx = /^[a-z][\d_a-z]*(\.[\d_a-z]+)+[\d_a-z]:\/\/[a-z]*$/;
```

Comment:
We should also allow valid URI symbols (e.g. `-` and `/`) after the `://`, as we have native apps that actually use this pattern:
```
io.logto://demo-app/callback
```